File: Data Structures & Algorithms/jump-game/submission-1.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        boolean[] can = new boolean[n];
        can[0]=true;
        for(int i=0;i<n;i++){
            if(can[i]){
                for(int j=1;j<=nums[i];j++){
                    if(i+j<n)can[i+j]=true;
                }
            }
        }
        return can[n-1];
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int> map;
        for(int i=0;i<static_cast<int>(nums.size());++i){
            int remaining = target - nums[i];
            if(map.find(remaining)==map.end()){
                map[nums[i]] = i;
                continue;
            }
            return {map[remaining], i};
        }
    }
};
